arroyo-api/src/main.rs

```diff
@@ -603,17 +577,39 @@ impl ApiGrpc for ApiServer {
         request: Request<CreatePipelineReq>,
     ) -> Result<Response<CreateJobResp>, Status> {
         let (request, auth) = self.authenticate(request).await?;
-        self.start_or_preview(request.into_inner(), false, auth)
+
+        let mut client = self.client().await?;
+        let transaction = client.transaction().await.map_err(log_and_map)?;
+        transaction
+            .execute("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE", &[])
             .await
+            .map_err(log_and_map)?;
+
+        let resp = start_or_preview(request.into_inner(), false, auth, &transaction).await;
+
+        transaction.commit().await.map_err(log_and_map)?;
+
+        resp
     }
 
     async fn preview_pipeline(
         &self,
         request: Request<CreatePipelineReq>,
     ) -> Result<Response<CreateJobResp>, Status> {
         let (request, auth) = self.authenticate(request).await?;
-        self.start_or_preview(request.into_inner(), true, auth)
+
+        let mut client = self.client().await?;
+        let transaction = client.transaction().await.map_err(log_and_map)?;
+        transaction
+            .execute("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE", &[])
             .await
+            .map_err(log_and_map)?;
+
+        let resp = start_or_preview(request.into_inner(), true, auth, &transaction).await;
+
+        transaction.commit().await.map_err(log_and_map)?;
+
+        resp
```

Comment:
TODO: same logic as above, with the exception of the boolean "preview" parameter.  Refactor into one function.

Reply:
See #69 for more.